In [2]:
import numpy as np
import pandas as pd
import seaborn as sns

다음 조건을 만족하는 승객의 나이 평균과 조건을 만족하는 데이터의 개수를 구하세요.

fare를 30 이상 40 미만 지불한 승객

pclass는 1등급

In [40]:
df = sns.load_dataset('titanic')
cond1 = (df['fare'] >= 30) & (df['fare'] < 40)
cond2 = df['pclass'] == 1

df_copy1 = df[cond1 & cond2]

df_copy1['age'].value_counts().sum()

21

In [104]:
df_copy2 = df.loc[cond1 & cond2, 'age'].count()
print(df_copy2)

21


In [42]:
df_copy1['age'].mean()

44.095238095238095

In [106]:
df.loc[cond1 & cond2, 'age'].mean()

44.095238095238095

In [44]:
diamond = sns.load_dataset('diamonds')
diamond

carat        cut color clarity  depth  table  price     x     y     z
0       0.23      Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1       0.21    Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2       0.23       Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3       0.29    Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4       0.31       Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75
...      ...        ...   ...     ...    ...    ...    ...   ...   ...   ...
53935   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75  5.76  3.50
53936   0.72       Good     D     SI1   63.1   55.0   2757  5.69  5.75  3.61
53937   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66  5.68  3.56
53938   0.86    Premium     H     SI2   61.0   58.0   2757  6.15  6.12  3.74
53939   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83  5.87  3.64

[53940 rows x 10 columns]

In [48]:
depth_min_value = diamond['depth'].min()
print(depth_min_value)


43.0


In [58]:
print(diamond['carat'].mean()) #  평균
print(diamond['carat'].var()) #  분산


0.7979397478680014
0.22468665982273753


mean    0.797940
var     0.224687
Name: carat, dtype: float64
=> 동시에 출력 방법은???

agg() 메서드
agg()는 pandas에서 제공하는 강력한 데이터 집계 메서드로, Series나 DataFrame에 
대해 여러 가지 집계 함수(aggregation function)를 한 번에 적용할 수 있도록 도와줍니다.


다중 집계 함수 지원:
한 번의 호출로 여러 함수를 적용할 수 있습니다.
대표적으로 mean, sum, min, max, var 같은 집계 함수가 사용됩니다.

Series와 DataFrame 모두에서 사용 가능:
Series에 적용하면 해당 열에 대한 집계를 수행합니다.
DataFrame에 적용하면 열별로 집계를 수행합니다.

사용자 정의 함수 지원:
내장된 함수뿐 아니라, 사용자가 작성한 함수를 전달할 수도 있습니다.




In [108]:
diamond['carat'].agg(['mean', 'var'])

mean    0.797940
var     0.224687
Name: carat, dtype: float64

x, y 합계 및 표준편차

In [70]:

result_data = [
    [diamond['x'].sum(), diamond['y'].sum()],
    [diamond['x'].std(), diamond['y'].std()]
]

result_df = pd.DataFrame(result_data, index=['sum', 'std'], columns=['x', 'y'])
result_df

x              y
sum  309138.620000  309320.330000
std       1.121761       1.142135

In [82]:

result_data = {
    'x': [diamond['x'].sum(), diamond['y'].sum()],
    'y': [diamond['x'].std(), diamond['y'].std()]
}

result_df = pd.DataFrame(result_data, index=['sum', 'std'])
result_df

x         y
sum  309138.62  1.121761
std  309320.33  1.142135

In [96]:
result_zip = zip(
    ['sum', 'std'],
    [diamond['x'].sum(), diamond['x'].std()],
    [diamond['y'].sum(), diamond['y'].std()]
)

result_df = pd.DataFrame(result_zip, columns=['index', 'x', 'y']).set_index('index')
result_df



x              y
index                              
sum    309138.620000  309320.330000
std         1.121761       1.142135

In [98]:
result_array = np.array([
    [diamond['x'].sum(), diamond['y'].sum()],
    [diamond['x'].std(), diamond['y'].std()]
])

result_df = pd.DataFrame(result_array, index=['sum', 'std'], columns=['x', 'y'])
print(result_df)

                 x              y
sum  309138.620000  309320.330000
std       1.121761       1.142135


In [110]:
diamond[['x', 'y']].agg(['sum', 'std'])

x              y
sum  309138.620000  309320.330000
std       1.121761       1.142135

In [72]:
penguin = sns.load_dataset('penguins')
penguin

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    Male  
1         3800.0  Female  
2         3250.0  Female  
3            NaN     NaN  
4         3450.0  Female  
..           ...     ...  
339          NaN     NaN  
340       4850.0  Female  
341       5750.0    Male  
342       5200.0  Female  
343       5400.0    Male  

[344 rows x 7 columns]

species 컬럼의 고유값

In [74]:
penguin['species'].unique()

array(['Adelie', 'Chinstrap', 'Gentoo'], dtype=object)

island 컬럼의 최빈값을 출력해 주세요

In [78]:
penguin['island'].mode()

0    Biscoe
Name: island, dtype: object

body_mass_g 컬럼의 10% 분위수 값(하위 10%)을 출력해 주세요

In [114]:
# ???
penguin['body_mass_g'].quantile(0.1)

3300.0

body_mass_g 컬럼의 80% 분위수 값(상위 20%)을 출력해 주세요

In [118]:
# ???
penguin['body_mass_g'].quantile(0.8)

4950.0